kiosk/sources/extensions/marketplace_adapter.move

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// The best practical approach to trading on marketplaces and favoring their
+/// fees and conditions is issuing an additional `TransferRequest` (eg `Market`).
+/// However, doing so is not always possible because it must be copied from
+/// TransferRequest<Item>. Mainly because the price of the sale is not known to
+/// the very moment of the sale. And if there's already a TransferRequest<Item>,
+/// how do we enforce the creation of an extra request? That means that sale has
+/// already happened.
+///
+/// To address this problem and also solve the extension interoperability issue,
+/// we created a `marketplace_adapter` - simple utility which wraps the
+/// `PurchaseCap` and handles the last step of the purchase flow in the Kiosk.
+///
+/// Unlike `PurchaseCap` purpose of which was to be "free", `MarketPurchaseCap`
+/// - the wrapper - only comes with a `store` to reduce the amount of scenarios
+/// when it is transferred by accident or sent to an address / object.
```

Comment:
Confused by this, why does only having `store` help reduce accidental transfers -- someone is free to wrap the type and then transfer however they like, right?

Reply:
Then it's very explicit that the Cap needs to be wrapped. Not having `key` makes you think, and that's exactly what we need.